File: js/repositories/UnavailableScheduleRepository.js
/**
 * Dakar Bus - Groupe 11
 * UnavailableScheduleRepository : premier repository, aucune donnée fausse
 *
 * Comportement :
 * aucune source réelle branchée
 *   ↓
 * liste vide
 *   ↓
 * Horaire non disponible
 *
 * Il ne doit générer aucun horaire.
 * Cette étape est volontaire : architecture fonctionnelle avant première source réelle.
 */

import { ScheduleRepository } from './ScheduleRepository.js';

export class UnavailableScheduleRepository extends ScheduleRepository {
  /**
   * @param {string} stopId
   * @returns {Promise<Array>} toujours vide
   */
  async departuresForStop(stopId) {
    // Aucune source réelle branchée à ce stade
    // Retourne volontairement une liste vide, pas de génération
    return [];
  }
}

export default UnavailableScheduleRepository;

File: flutter-src/lib/repositories/unavailable_schedule_repository.dart
/// Dakar Bus - Groupe 11
/// UnavailableScheduleRepository : premier repository, aucune donnée fausse
/// Comportement :
/// aucune source réelle branchée -> liste vide -> Horaire non disponible
/// Il ne doit générer aucun horaire. Étape volontaire.

import '../models/departure.dart';
import 'schedule_repository.dart';

class UnavailableScheduleRepository implements ScheduleRepository {
  @override
  Future<List<Departure>> departuresForStop(String stopId) async {
    // Aucune source réelle branchée à ce stade
    return [];
  }

  @override
  Future<List<Departure>> departuresForStopAndLine(String stopId, String lineId) async {
    final all = await departuresForStop(stopId);
    return all.where((d) => d.lineId == lineId).toList();
  }
}
